cygnus-twitter/src/main/java/com/telefonica/iot/cygnus/sinks/TwitterHDFSSink.java

```diff
@@ -0,0 +1,380 @@
+        } // try catch
+
+        super.start();
+    } // start
+
+    @Override
+    void persistBatch(TwitterBatch batch) throws Exception {
+        if (batch == null) {
+            LOGGER.debug("[" + this.getName() + "] Null batch, nothing to do");
+            return;
+        } // if
+
+        // get an aggregator for this destination and initialize it
+        HDFSAggregator aggregator = new JSONRowAggregator();
+
+        ArrayList<TwitterEvent> subBatch = batch.getBatch();
+        aggregator.initialize(hdfs_folder, hdfs_file);
+        for (TwitterEvent cygnusEvent : subBatch) {
+            aggregator.aggregate(cygnusEvent);
+        } // for
+
+        // persist the aggregation
+        persistAggregation(aggregator);
+
+
+    } // persistBatch
+
+    /**
+     * Class for aggregating aggregation.
+     */
+    private abstract class HDFSAggregator {
+
+        // string containing the data aggregation
+        protected String aggregation;
+        protected String hdfsFolder;
+        protected String hdfsFile;
+
+        public HDFSAggregator() {
+            aggregation = "";
+        } // HDFSAggregator
+
+        public String getAggregation() {
+            return aggregation;
+        } // getAggregation
+
+        public String getFolder(boolean enableLowercase) {
+            if (enableLowercase) {
+                return hdfsFolder.toLowerCase();
+            } else {
+                return hdfsFolder;
+            } // if else
+        } // getFolder
+
+        public String getFile(boolean enableLowercase) {
+            if (enableLowercase) {
+                return hdfsFile.toLowerCase();
+            } else {
+                return hdfsFile;
+            } // if else
+        } // getFile
+
+        public void initialize(String hdfs_folder, String hdfs_file) throws Exception {
+            hdfsFolder = hdfs_folder;
+            hdfsFile = hdfsFolder + "/" + hdfs_file;
+        } // initialize
+
+        public abstract void aggregate(TwitterEvent cygnusEvent) throws Exception;
+
+    } // HDFSAggregator
+
+    /**
+     * Class for aggregating batches in JSON row mode.
+     */
+    private class JSONRowAggregator extends HDFSAggregator {
+
+        @Override
+        public void initialize(String hdfs_folder, String hdfs_file) throws Exception {
+            super.initialize(hdfs_folder, hdfs_file);
+        } // initialize
+
+        @Override
+        public void aggregate(TwitterEvent cygnusEvent) throws Exception {
+
+            // get the event data (tweet)
+            String eventData = cygnusEvent.getEventData();
+
+            if (eventData == null || eventData.isEmpty()) {
+                LOGGER.warn("No event data (tweet), nothing is done ");
+                return;
+            } // if
+
+            aggregation += tweet2row(eventData);
+        } // aggregate
+
+        private String tweet2row(String eventData) {
+            return eventData.replaceAll("\n", "") + "\n";
```

Comment:
Since this seems to be a largely used method, I'll try to precompile the pattern. The "replaceAll" method compiles the given regular expression each time it is called. Neverhteless, precompiling a pattern implies just one compilation.